### PR TITLE
Added prev-next navigation to nested menus

### PR DESCRIPTION
**Pull request by pbberlin**
Nested menus allow for elegant organization of articles for one topic.

Docu already provides a nifty template for rendering nested menu:
https://gohugo.io/extras/menus/#rendering-menus:4044c7430755cd64e5b7c48e836cea57

Now I add two methods for previous and next article according to menu.
This is not trivial, because we have to step up or down on the levels of the menu tree.
Implementation via template functions infeasible.

Thus, I added methods, tests and docu.

Regards
  Peter

### CODE REVIEW

**Comment by bep** [comment]
Looks very solid to me, and will make the menu system very powerful for navigation.

File: hugolib/menu.go [hunk 1]
```diff
@@ -181,3 +181,150 @@ func (m Menu) Reverse() Menu {
...
+// The resulting prev/next menu menu entries are suitable for
+// traversing a menu hierarchy exactly like a word-processing software would:
+// 1.
+// 1.1
+// 1.2
+// 2.1
+// 2.1.1
+// 2.2
+//
+func (m Menu) hasMenuCurrentPrevNext(me *MenuEntry) (bool, *MenuEntry, *MenuEntry) {
```
Comment (pbberlin) [inline hugolib/menu.go]: I did not find a more descriptive method name

**Comment by bep** [comment]
@vjeantet you have been working with menus more than me. Is this a "very useful" feature?

**Comment by bep** [comment]
And @pbberlin I'm sorry I've been neglecting these old PRs. Not an excuse, but I believe this precedes me taking over as the lead maintainer of Hugo.

**Comment by vjeantet** [comment]
Yes this feature is really useful.

I implemented a template for this in my docdock theme :
(The old one was for nested menu)
(The new one works with nested sections)

**Comment by bep** [comment]
OK, I will have to think a little about this one ...

**Pull request closed without merging**